components/FileUploadField.tsx

```diff
@@ -0,0 +1,50 @@
+import { AttachmentIcon } from '@chakra-ui/icons';
+import { Button } from '@chakra-ui/react';
+import { SyntheticEvent, useRef } from 'react';
+
+// File upload button props
+type FileUploadProps = {
+  currentFile: File | null; // currently uploaded file
+  onUploadFile: (selectedFile: File) => void; // handle file upload
+};
+
+//TODO: rename this better??
+/**
+ * File upload button component allowing users to upload one file
+ * @param props - props
+ * @returns File upload button component allowing users to upload one file
+ */
+export default function FileUpload(props: FileUploadProps) {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+
+  const handleFileUpload = () => {
+    if (hiddenFileInput.current) {
+      hiddenFileInput.current.click();
+    }
+  };
+
+  const handleChange = (event: SyntheticEvent) => {
+    const target = event.target as HTMLInputElement;
+    const fileUploaded = (target.files as FileList)[0];
+    props.onUploadFile(fileUploaded ?? null);
+  };
```

Comment:
```suggestion
  const handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
    const target = event.target;
    const fileUploaded = (target.files)[0];
    props.onUploadFile(fileUploaded ?? null);
  };
```
I believe we can use type generics here to avoid type casting

Reply:
Actually I just realized that we're getting some linting errors if we remove the casting for some reason. Will leave it as is for now